src/MiddlewareFactory.php

```diff
@@ -87,21 +88,21 @@ private function isMiddlewareClassDefinition(array|callable|string $definition):
     }
 
     /**
-     * @psalm-assert-if-true array{0:class-string, 1:non-empty-string}|Closure $definition
+     * @psalm-assert-if-true array{0:class-string|object, 1:non-empty-string}|callable-object|Closure $definition
      */
     private function isCallableDefinition(array|callable|string $definition): bool
     {
-        if ($definition instanceof Closure) {
+        if ($definition instanceof Closure || is_object($definition)) {
```

Comment:
```suggestion
        if (is_callable($definition)) {
```

Object may not be callable.

Reply:
Hmm... the method signature only allows the callable object?

Reply:
Not only. Any callable: `[$object, 'method']`, `'anyFunctionName'`, object with `__invoke`, etc.